x-pack/plugins/enterprise_search/public/applications/app_search/components/documents/types.ts

```diff
@@ -0,0 +1,11 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+export interface FieldDetails {
+  name: string;
+  value: string | string[];
```

Comment:
[not a change request, just me thinking out loud] Someday in a very distant future it might be nice to clean that up a bit more in a server transform - e.g. by doing `isArray(value) ? value : [value]` we know for sure we're always dealing w/ arrays

Reply:
💯 Yeah that would be ideal

Reply:
I made a simple fix for this at the Logic level for now. This could be refactored later, but I think it's worth at least this small amount of effort because it will greatly simplify things downstream in the UI: 772e801f0863d6915b13ee91de74883365cb1f38

Reply:
Oh awesome, love it! I have a very slight preference to storing / testing `normalizeFields` in a `utils` file to separate out the array logic from `getDocumentDetails` test (in theory a unit test is easier to follow if it handles 1 thing at a time), but it's a very slight preference and I'm fine with merging as-is.

Reply:
I reverted it. We actually need to know if it's an array or not on the front-end.

https://github.com/elastic/kibana/pull/83324/commits/a40fcb9878d0f6d71281046c9cdcee4622870c2c

Reply:
Ahh no worries, I was hoping something like that wouldn't come up, but alas. Some day